### ISSUE TITLE
Code Snippets are not accessible 

### ISSUE THREAD

**Issue by ewaDev** [forem/forem#10335]
**Describe the bug**
On desktop, user cannot navigate to the code snippet using keyboard only. User cannot put the snippet in focus, cannot scroll through it or have it read by 

**To Reproduce**

1. Create a new post.
2. Insert a code snippet with a long line - for example : 
```JavaScript
console.log("Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
```
3. Publish.
4. On desktop try do navigate to the snippet using keyboard. You can't.
5. Attempt to scroll through to the end of the snippet - you can't.
6. This also means that the code snippets are not read by a screen reader.

**Expected behavior**

1. User should be able to navigate to a code snippet using a keyboard
2. User should be able to interact with the code snippet using the keyboard
3. Screen Reader should pick up 

**Screenshots**

Not needed.

**Desktop (please complete the following information):**

- OS, version:
- Browser, version:

**Smartphone (please complete the following information):**

- Device: Dell Laptop
- OS, version: Windows
- Browser, version: Version 85.0.4183.102 (Official Build) (64-bit)

**Additional context**

**Comment by Amorpheuz** [comment]
Hey @rhymes, I would love to give this a shot if it's still up for grabs.

**Comment by citizen428** [comment]
Hi @Amorpheuz, please don't ping individual team members unless it's part of an ongoing discussion. rhymes is off this week. I'll assign you the issue now.

**Comment by Amorpheuz** [comment]
Oh, Sorry folks. Thanks for the help citizen.

**Comment by citizen428** [comment]
This issue is up for grabs again. If you want to work on it it's probably advisable to have a look at the closed PR further up and read some of the comments there.

**Comment by andreamartz** [comment]
@nickytonline and @cmgorton I'd like to take this issue.

**Comment by andreamartz** [comment]
I'm sorry. Can I be unassigned this issue? I've overcommitted.

**Comment by cmgorton** [comment]
No problem @andreamartz . I'll unassign you.

**Comment by moopet** [comment]
As far as the scrolling is concerned, I've noticed that in Firefox/Windows, no horizontal scrollbar appears until you focus or hover the code area, wherease in Chrome/Windows or Firefox/Chrome/Safari on Mac/Arch it's fine.

Without a scrollbar, there's no indication that code goes over the line, and with many people folding code across lines it can be misleading.

**Comment by mikeydorje** [comment]
Closing as stale.

Accessibility bug for code snippets not supporting keyboard navigation. Multiple attempts were made to fix it (Amorpheuz's PR #11058, andreamartz's assignment), but neither resulted in a solution. Last activity was May 2022 discussing scrollbar visibility issues.

If you're experiencing this issue, please reopen with context.
